src/network.rs

```diff
@@ -8,14 +8,18 @@ use url::Url;
 
 use crate::errors::{Error, OptionExt};
 
+#[derive(Debug, Serialize)]
+pub struct RpcConfig {
+    pub url: String,
+    pub cred: Option<(String, String)>, // (username, password)
+    pub cookie: Option<String>,
```

Comment:
The bitcoincore-rpc crate actually has an enum over these two variants :p It's a nit, though.